custom_components/intesisbox/intesisbox.py

```diff
@@ -244,17 +268,37 @@ def set_horizontal_vane(self, vane: str):
     def _set_value(self, uid: str, value: str | int) -> None:
         """Change a setting on the thermostat."""
         try:
-            self._write(f"SET,1:{uid},{value}")
+            # self._write(f"SET,1:{uid},{value}")
+            asyncio.run(self._writeasync(f"SET,1:{uid},{value}"))
         except Exception as e:
             _LOGGER.error("%s Exception. %s / %s", type(e), e.args, e)
 
-    def set_mode(self, mode: str) -> None:
-        """Change the thermostat mode (heat, cool, etc)."""
-        if not self.is_on:
-            self.set_power_on()
-
+    def set_mode(self, mode):
+        """Send mode and confirm change before turning on."""
+        """ Some units return responses out of order"""
+        _LOGGER.debug(f"Setting MODE to {mode}.")
         if mode in MODES:
             self._set_value(FUNCTION_MODE, mode)
+        if not self.is_on:
+            """ "Check to ensure in correct mode before turning on"""
+            retry = 30
+            while self.mode != mode and retry > 0:
```

Comment:
I think the intention of this change makes sense, I'm just a bit unsure about the approach.

The switch to spawning a new tasks in _set_value and here feels unpredictable from a sequencing perspective.
The code might be much cleaner if you added an `async set_mode` function that contained all the logic and replaced the non-async function with a `asyncio.run(self._async_set_mode(...))` 

This would allow you to directly await the other co-routines and keep the execution sequential.

Reply:
This concept makes sense, the implementation is above my level of understanding sorry.  The key point here that solved all my issues is set the MODE first, confirm its set, the send the ON command.  The 3 AC units and IB's all have various, inconsistent response times and more often than not send the On response is sent from the IB before the MODE response, even if MODE is is sent by client a second later.  Very strange, like the priority in CHN response is for the ONOFF over MODE.

Reply:
No worries, we can leave it as is and clean this up later if it feels worth someones time.